src/BBT.Maybe/NoneCase.cs

```diff
@@ -35,7 +35,7 @@ internal NoneCase(bool isNone)
         /// <param name="b">Maybe to compare.</param>
         public static bool operator !=(NoneCase a, NoneCase b)
         {
-            return a != b;
+            return !(a == b);
```

Comment:
Is this change intended?

Reply:
yes, recursive call of operator causes stack overflow exception.